src/expr/src/linear.rs

```diff
@@ -1287,11 +1286,16 @@ pub mod plan {
         pub fn permute(&mut self, permutation: &[usize]) {
             let (map, new_arity) = permutation_to_map_and_new_arity(permutation);
             self.mfp.mfp.permute(map, new_arity);
+            let permutation = permutation
```

Comment:
Alternatively, we could make `permute` return the `shuffle` map and use `permute_map` instead for the bounds.

In any case, I think we should mention here that the bounds may reference internal CSE columns added by M part of the MFP, which seems to be missed detail that led to this bug.

Reply:
Another alternative approach would be to inline the CSEs when extracting the bounds in `create_from`, to avoid having some client code of the MFP relying on its internal state, although it leads to avoidable computation. Just throwing the idea out.